### PR TITLE
Feature: Hardware check

### PR DESCRIPTION
**Pull request by ghost**
Checks user hardware and halts execution if it doesn't match minimum requirements (defined in ./src/hardware.rs)

### CODE REVIEW

**Comment by joshbenaron** [comment]
1) If you want review you need to request. 2) Can we upstream merge this with master

**Comment by joshbenaron** [comment]
Also default port for validator is now 42069 (docker-compose is using something else)

**Comment by ghost** [comment]
All good now

**Comment by joshbenaron** [comment]
I will fix

File: src/hardware.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+use sysinfo::{System, SystemExt};
+
+pub const MIN_RAM_KB : u64 = 524800;    //512mb
```
Comment (joshbenaron) [inline src/hardware.rs]: Set to 4 GB please

**Review by joshbenaron — APPROVED** [review]
LGTM

File: src/hardware.rs (hunk 1 above)
Comment (joshbenaron) [reply src/hardware.rs]: @bluekirby1111

**Comment by ghost** [comment]
Once signing is ok, we can merge this one

File: src/hardware.rs (hunk 1 above)
Comment (ghost) [reply src/hardware.rs]: Done

**Pull request merged**